### ISSUE TITLE
fitsio failing to build wheel for pip based install on github.

### ISSUE THREAD

**Issue by eacharles** [esheldon/fitsio#414]
It looks like the specific error is:
configure: error: Unable to locate bz2 library needed when enabling bzip2 support; try specifying the path

You can find an example here:
https://github.com/LSSTDESC/rail_pipelines/actions/runs/12282370023/job/34273426914

**Comment by esheldon** [comment]
Can you try installing libbz2 in the container?

**Comment by esheldon** [comment]
Probably something like `sudo apt install libbz2-dev`

**Comment by esheldon** [comment]
I can give a little context:  pip cannot install the bzip2 library, so it is up to the user to make sure it is available. My guess is the github containers may no longer be including that library as part of the standard set of installations.

**Comment by eacharles** [comment]
That does work, but we have something like 60 or so workflows, and I'd rather not have to update them all.

**Comment by eacharles** [comment]
That's like 4-5 workflows per package across somethink like 12-15 packages.

**Comment by esheldon** [comment]
This is out of my control.  Neither I or the fitsio package are affiliated with github

I think you have a few options:

1) Complain to github.
2) edit the workflow to install the dependency
3) edit the workflow to choose a different container.  There are alternative containers out there that may have your dependency

**Comment by rmjarvis** [comment]
FWIW, I worked around this issue by adding the following to my ci.yml file:
```
- name: Install libbz2
  if: matrix.os == 'ubuntu-latest'
  run: sudo -H apt-get install -y libbz2-dev
```
This seems to have worked for me.

**Comment by ntessore** [comment]
Hi @esheldon, have you thought about providing wheels for fitsio? The process is pretty much automatic with cibuildwheel ([here an example of mine](https://github.com/ntessore/healpix/blob/main/.github/workflows/cibuildwheel.yml))

**Comment by esheldon** [comment]
@beckermr do you have ideas about this proposal?

**Comment by ntessore** [comment]
IIUC cibuildwheel will take care of the shared libraries using auditwheel/delocate

**Comment by rmjarvis** [comment]
I think that's right.  GalSim uses it and it works correctly with the FFTW library.

**Comment by rmjarvis** [comment]
You can use our wheel.yml as a template if you want:
https://github.com/GalSim-developers/GalSim/blob/releases/2.6/.github/workflows/wheels.yml

**Comment by beckermr** [comment]
Yes if people want wheels, we can ship them. We still need to ship the sdist as well. I do not like wheels but lots of folks do. Best to give the people what they want.

**Comment by beckermr** [comment]
:100: that's correct @rmjarvis :)

**Comment by beckermr** [comment]
Wheels are [building](https://github.com/esheldon/fitsio/pull/416)! I need to add tests of them and then the PR is ready to go!

**Comment by esheldon** [comment]
Awesome, thanks Matt

**Comment by beckermr** [comment]
Wheels for version 1.2.5 are now on pypi. Let us know if you have issues!
